### PR TITLE
Add tagged property to Tool

### PR DESCRIPTION
**Pull request by Jbcampbe**
Client requested ability to track wether a tool has been tagged or not

### CODE REVIEW

**Comment by Jbcampbe** [comment]
Ready for Review

**Comment by ekraihan** [comment]
> Ready for Review



> Ready for Review

I don't see the ADD COLUMN command added? Maybe you didn't push your most recent commit?

**Comment by Jbcampbe** [comment]
@ekraihan Comments Addressed

File: sql/schema/tool.sql [hunk 1]
```diff
@@ -24,6 +23,9 @@ CREATE TABLE IF NOT EXISTS public.tool (
 CREATE UNIQUE INDEX IF NOT EXISTS tool_id_index ON public.tool (id);
 CREATE INDEX IF NOT EXISTS tool_filters_index ON public.tool (type_id, brand_id, owner_id, status);
 
+-- Add "tagged" column to tools setting the tools that already exist to true (all tools in the production app had been tagged)
+ALTER TABLE public.tool ADD COLUMN IF NOT EXISTS tagged BOOL NOT NULL DEFAULT true;
```
Comment (ekraihan) [inline sql/schema/tool.sql]: Hm, I'm not sure I like setting every tool to true here... Just from a design perspective. Also, this sets the default for any future tools being added to true as well.

File: sql/schema/tool.sql [hunk 2]
```diff
@@ -23,6 +23,9 @@ CREATE TABLE IF NOT EXISTS public.tool (
 CREATE UNIQUE INDEX IF NOT EXISTS tool_id_index ON public.tool (id);
 CREATE INDEX IF NOT EXISTS tool_filters_index ON public.tool (type_id, brand_id, owner_id, status);
 
+-- Add "tagged" column to tools setting the tools that already exist to true (all tools in the production app had been tagged)
+ALTER TABLE public.tool ADD COLUMN IF NOT EXISTS tagged BOOL NOT NULL DEFAULT false;
```
Comment (ekraihan) [inline sql/schema/tool.sql]: Could you move this up to line 23 to be more in the flow of table creation?

File: sql/views/tool-search-view.sql [hunk 3]
```diff
@@ -5,22 +5,23 @@ BEGIN;
...
+    COALESCE(public.user.first_name, '') AS user_first_name,
+    COALESCE(public.user.last_name, '') AS user_last_name,
+    tool.id::text AS tool_id,
+    brand.name AS brand_name,
+    type.name AS type_name,
+    tool.model_number AS tool_model_number,
+    tool.serial_number AS tool_serial_number,
+    COALESCE(public.location.name, '') AS location_name,
+    tool.status::text AS tool_status,
+    tool.tagged::text AS tagged
```
Comment (ekraihan) [inline sql/views/tool-search-view.sql]: This view is for fuzzy searching only. If you want to be able to fuzzy search on tagged,  you'll need to add the tagged field to the `GREATEST` clause in sql/functions/search-fuzzy-ids-tool.sql

Comment (ekraihan) [reply sql/views/tool-search-view.sql]: @Jbcampbe ^

Comment (Jbcampbe) [reply sql/views/tool-search-view.sql]: I don't think we will need to fuzzy search for true and false. I'll remove

File: graphql/schema/tool.graphql [hunk 4]
```diff
@@ -2,52 +2,56 @@
...
+  tagged:            Boolean
 }
 
 input ToolFilter {
   brand_ids:     [ID!]
   type_ids:      [ID!]
   tool_statuses: [InServiceToolStatus!]
-  owner_ids:     [ID!]
+  owner_ids:     [ID!],
+  tagged:        Boolean
```
Comment (ekraihan) [inline graphql/schema/tool.graphql]: Can also remove this

File: sql/schema/tool-snapshot.sql [hunk 5]
```diff
@@ -14,6 +14,7 @@ CREATE TABLE IF NOT EXISTS public.tool_snapshot (
 	serial_number 		str_t			 			 NOT NULL,
   owner_id          id_t             NOT NULL,
   owner_type        tool_owner_type  NOT NULL,
+  tagged            boolean          NOT NULL DEFAULT false,
```
Comment (ekraihan) [inline sql/schema/tool-snapshot.sql]: Need to remove this line

File: sql/functions/search-strict-tool.sql [hunk 6]
```diff
@@ -20,41 +20,36 @@ CREATE OR REPLACE FUNCTION retina.search_strict_tool (
...
-      query = query || ' AND tagged = $5';
+      queryString = queryString || ' AND tagged = ' || tagged;
     END IF;
 
-    query = query ||
-      ' ORDER BY tool.id ASC' ||
-      ' OFFSET $6' ||
-      ' LIMIT $7';
-
-    RETURN QUERY EXECUTE query USING owner_ids,
```
Comment (ekraihan) [inline sql/functions/search-strict-tool.sql]: I changed some code to avoid using this `USING` list

File: sql/functions/helpers.sql [hunk 7]
```diff
@@ -0,0 +1,15 @@
+
+/**
+ * @function array_to_string_list
+ *
+ * @param   array     An array of any type
+ * @returns The concatenation of the string form of all the elements in the array,
+ *          separated by commas, and each wrapped with single quotes
+ */
+CREATE FUNCTION retina.array_to_string_list("array" anyarray) RETURNS character varying
```
Comment (ekraihan) [inline sql/functions/helpers.sql]: Helper for changes in `search-strict-tool.sql`

File: tests/test-db.js [hunk 8]
```diff
@@ -1170,6 +1171,16 @@ describe('Database creation and usage', async () => {
     });
   });
 
+  describe('array_to_string_list()', () => {
+    // TODO: Had to use a raw sql query because postgres wasn't able to derive
+    //       the type of the js array being passed. Possibly add logic to
+    //       executeDbFunction that explicitly converts js arrays to "ARRAY[x,y,z]"?
+    it('combines array values into a list of quoted values', async () => {
```
Comment (ekraihan) [inline tests/test-db.js]: Test for helper

**Pull request merged**